src/stores/plan.ts

```diff
@@ -51,16 +51,30 @@ export const planMergeRequestsIncoming = gqlSubscribable<PlanMergeRequest[]>(
   gql.SUB_PLAN_MERGE_REQUESTS_INCOMING,
   { planId },
   [],
-  (planMergeRequests: PlanMergeRequestSchema[]): PlanMergeRequest[] =>
-    planMergeRequests.map(planMergeRequest => ({ ...planMergeRequest, pending: false, type: 'incoming' })),
+  (planMergeRequests: PlanMergeRequestSchema[]): PlanMergeRequest[] => {
+    const requests: PlanMergeRequest[] = planMergeRequests.map(planMergeRequest => ({
+      ...planMergeRequest,
+      pending: false,
+      type: 'incoming',
+    }));
+    requests.sort((a, b) => (a.id > b.id ? -1 : 0)); // Sort by descending time order
```

Comment:
Instead of sorting in JavaScript I think we can just sort in the query. For example:

```gql
subscription SubPlanMergeRequestsIncoming($planId: Int!) {
  merge_request(where: { plan_id_receiving_changes: { _eq: $planId } }, order_by: { id: desc }) {
    id
    plan_receiving_changes {
      id
      name
    }
    plan_snapshot_supplying_changes {
      name
      snapshot_id
    }
    requester_username
    reviewer_username
    status
  }
}
```

Reply:
Oh excellent I'll give that a go

Reply:
Done